### ISSUE TITLE
OpenWRT / LEDE support

### ISSUE THREAD

**Issue by renekliment** [alexa-pi/AlexaPi#135]
We're **kind of** close, so opening this ticket. 

What we need to do / issues with this:
- [ ] have SoX playback handler (#106), because that is the only doable way to play music on this OS (apart from mpd, which is for different stuff, although it could be used in theory); SoX will be default for AlexaPi, so this is totally fine
- probably adjust scripts, because the shell / commands there are not full-featured
    - [ ] setup script
    - [ ] init script - probably a custom one
- [ ] does `pip` work as expected on OpenWRT? some packages need compiling stuff
- vad (webrtcvad): this is fun stuff ... tried compiling under PowerPCSPE (router Turris 1.0) ... that's a no-go; I'm not sure about other architectures; webrtcvad is hardcoded and not a module we can just not use, so we might need to either make it a module somehow or do patches for OpenWRT support
- triggers - this is a very tricky one:
    - snowboy: this depends on the architecture and I'm not sure the snowboy lib is compiled against the right stuff so it might not work under OpenWRT, which uses a different C library than full distros 
    - pocketsphinx - another funny story - I believe it requires some compilation and the right architecture maybe?
    - platform - well this depends of course:
        - GPIO - there are boards which use a variation on the RPi.GPIO, but if we implement the general sysfs/GPIO (#37) for crying out loud, we have no issues here
        - other platforms like `serialplatform` - if you have a router with no GPIO / exposed, but it has an USB, you can use your Arduino for triggering and indicating stuff

So to sum up, AlexaPi on OpenWRT is doable, if you have the right platform (device, architecture). Some issues (like being able to compile stuff) can be solved by actually running Debian under LXC on OpenWRT, but there lies another funny story - in order to be able to do sound, you'd have to route the sound through PulseAudio from the container to the host.
